R/makeTreeSummarizedExperimentFromBiom.R

```diff
@@ -112,17 +125,20 @@ makeTreeSEFromBiom <- function(
         rownames(feature_data) <- rownames(counts)
     # Otherwise convert it into correct format if it is a list
     } else if( is(feature_data, "list") ){
+        # Clean feature_data from possible character artifacts
+        feature_data <- .detect_taxa_artifacts_and_clean(feature_data,
+                                                         cleanTaxaPattern)
         # Feature data is a list of taxa info
         # Get the maximum length of list
         max_length <- max( lengths(feature_data) )
-        # Clean feature_data from possible character artifacts
-        feature_data <- lapply(
-            feature_data,
-            gsub, pattern = cleanTaxaPattern, replacement = "")
+        
+        # In case the list of taxa info have different lengths
+        if (!all(lengths(feature_data)==max_length)){
         # Get the column names from the taxa info that has all the levels that occurs
         # in the data
         colnames <- names( head( feature_data[ lengths(feature_data) == 
                                                    max_length ], 1)[[1]] )
```

Comment:
the "head" seems unnecessary here?

Reply:
Honestly, it was existing in the implementation. The [`feature_data <- biomformat::observation_metadata(obj)`](https://github.com/microbiome/mia/blob/issue_303/R/makeTreeSummarizedExperimentFromBiom.R#L98) might result in a dataframe or a list. But the structure of the list return was not clear to me. Following the rest of the original code after its loading; I understood that it's just a list of taxonomies present. I will try look for some example online when it's loaded as a list to understand the structure better; it was not very clear to me neither :/

Reply:
I just mean that the result is equivalent with or without "head()". If it does not add anything you can remove the head() part from the code?

Reply:
I agree, I just need if it's ok, to test it with some real file example with `feature_data` being loaded as a list. So far when I have been developing and testing it as a list;  I was just using the `extdata/testdata/Aggregated_humanization2.biom` which returns  `feature_data` as a dataframe. And for the sake of testing I was manually casting it into a list to check that the part of the code that processes `feature_data` as a list is working. Or shall I just remove the `head` part and test it's behavior by similarly casting that `feature_data` as a list?

Reply:
Whatever works. My comment only meant that the head() in that call seemed irrelevant to have.

Reply:
At that part [of the code ](https://github.com/microbiome/mia/blob/issue_303/R/makeTreeSummarizedExperimentFromBiom.R#L139) it was originally assumed that the resulting `feature_data` is a list with unequal lengths.
For testing; so I used the same testing procedure mentioned above (casting the same `feature_data` from dataframe to list), however I changed `feature_data` as a list in a way the lengths are unequal. Then run the code that handle `feature_data` as a list (results at screenshot). it seems using: 
`colnames <- names( head( feature_data[ lengths(feature_data) ==max_length ], 1)[[1]] )`
or
`colnames <- names( feature_data[ lengths(feature_data) ==max_length ][[1]] )`
Are producing null as colnames. But rather just calling `colnames <- names(feature_data)` would do the job; only if the originally loaded `feature_data` as a list is named.
![image](https://github.com/microbiome/mia/assets/86964562/8795530f-a055-45ab-82ad-7e2c66dddf67)

Reply:
I was just thinking, if in the case that the `feature_data` loaded from a biom file with unequal vectors whiting it (supposing that always the list structure is the same; a vector for each taxonomy rank in that list) but with no names.
Could the following for example be safer?:
 
![image](https://github.com/microbiome/mia/assets/86964562/093574c6-037f-4c79-aa43-81e9b9de82c0)

Reply:
seems feasible